onnxscript/rewriter/pattern_test.py

```diff
@@ -323,6 +323,39 @@ def double(op, x):
         self.assertEqual(count, 1)
         self.assertEqual(model.graph.opset_imports["custom.domain"], 10)
 
+    def test_opset_import_in_function(self):
+        def add_same(x):
+            return x + x
+
+        def double(op, x):
+            return op.Double(x, domain="custom.domain", version=10)
+
+        rule = pattern.RewriteRule(add_same, double)
+
+        model_proto = onnx.parser.parse_model(
+            """
+            <ir_version: 7, opset_import: [ "" : 17, "pkg.custom": 1]>
+            agraph (float[N] x) => (float[M] z)
+            {
+                z = afunction (x)
+            }
+            <domain: "pkg.custom", opset_import: [ "" : 17]>
+            afunction (x) => (z)
+            {
+                y = Add (x, x)
+                z = Relu (y)
+            }
+        """
+        )
+        model = ir.serde.deserialize_model(model_proto)
+        count = pattern.RewriteRuleSet([rule], commute=True).apply_to_model(model)
+        self.assertEqual(count, 1)
+        self.assertEqual(len(model.functions), 1)
+        self.assertEqual(model.graph.opset_imports["custom.domain"], 10)
+        self.assertEqual(
+            model.functions[("pkg.custom", "afunction", "")].opset_imports["custom.domain"], 10
+        )
```

Comment:
I think it is a good idea to deserialize model and call checker ... Just realizing this after Xavier's bug reports, it will help us catch other bugs.

Reply:
I tried but we would get this instead:

```
onnxscript/rewriter/pattern_test.py:360: in test_opset_import_in_function
    onnx.checker.check_model(model_ir)
../anaconda3/envs/main-dev/lib/python3.11/site-packages/onnx/checker.py:179: in check_model
    C.check_model(
E   onnx.onnx_cpp2py_export.checker.ValidationError: No Op registered for afunction with domain_version of 17
```

It lookes like checker does not like the idea of ("", 17)? Or it's complaining about we delete the used opset?

Reply:
I will raise an issue about adding checker into optimizer/rewriter.

Reply:
Oh, line 340 should be `z = pkg.custom.afunction(x)`

Reply:
I will do a follow up